javatests/arcs/core/storage/CapabilitiesResolverTest.kt

```diff
@@ -0,0 +1,140 @@
+        assertThat(resolver1.createStorageKey(Capabilities.Persistent, "abc012") is DatabaseStorageKey).isTrue()
+
+        CapabilitiesResolver.reset()
+        val resolver2 = CapabilitiesResolver(options)
+        assertThat(resolver2.createStorageKey(Capabilities.TiedToArc) is VolatileStorageKey).isTrue()
+        assertThrows(IllegalStateException::class) {
+            resolver2.createStorageKey(Capabilities.TiedToRuntime)
+        }
+    }
+
+    @Test
+    fun capabilitiesResolver_staticCreators() {
+        assertThat(CapabilitiesResolver.defaultCreators.size).isEqualTo(1);
+        assertThat(CapabilitiesResolver.registeredCreators.isEmpty()).isTrue()
+
+        CapabilitiesResolver.registerDefaultKeyCreator(
+            "test1",
+            Capabilities.TiedToRuntime,
+            {(arcId), _ -> RamDiskStorageKey(arcId.toString())}
+        )
+        assertThat(CapabilitiesResolver.defaultCreators.size).isEqualTo(2);
+        assertThat(CapabilitiesResolver.registeredCreators.isEmpty()).isTrue()
+
+        CapabilitiesResolver.registerKeyCreator(
+            "test2",
+            Capabilities.Persistent,
+            {(arcId), _ -> RamDiskStorageKey(arcId.toString())}
+        )
+        assertThat(CapabilitiesResolver.defaultCreators.size).isEqualTo(2);
+        assertThat(CapabilitiesResolver.registeredCreators.size).isEqualTo(1)
+
+        CapabilitiesResolver.reset()
+        assertThat(CapabilitiesResolver.defaultCreators.size).isEqualTo(2);
+        assertThat(CapabilitiesResolver.registeredCreators.isEmpty()).isTrue()
+    }
+
+//    @Test
+//    fun capabilitiesResolver_findsProtocolsForCapabilities() {
+//        val options = CapabilitiesResolver.StorageKeyOptions(ArcId.newForTest("test"))
+//        val resolver1 = CapabilitiesResolver(options)
+//        assertThat(resolver1.findStorageKeyProtocols(Capabilities.TiedToArc)).containsExactly("volatile")
+//        assertThat(resolver1.findStorageKeyProtocols(Capabilities.TiedToRuntime).isEmpty()).isTrue()
+//        assertThat(resolver1.findStorageKeyProtocols(Capabilities.Persistent).isEmpty()).isTrue()
+//
+//        assertThat(CapabilitiesResolver.registeredCreators.size).isEqualTo(0)
+//        RamDisk.clear()
+//        assertThat(CapabilitiesResolver.registeredCreators.size).isEqualTo(1)
+////        DatabaseDriverProvider.configure(MockDatabaseFactory(), mapOf<String, Schema>()::get)
+//        val resolver2 = CapabilitiesResolver(options)
+//        assertThat(resolver2.findStorageKeyProtocols(Capabilities.TiedToArc)).containsExactly("volatile")
+//        assertThat(resolver2.createStorageKey(Capabilities.TiedToRuntime) is RamDiskStorageKey).isTrue()
+//        assertThat(resolver2.findStorageKeyProtocols(Capabilities.TiedToRuntime)).containsExactly("ramdisk")
+////        assertThat(resolver2.findStorageKeyProtocols(Capabilities.Persistent)).containsExactly("db")
+//    }
```

Comment:
Can this be deleted? Or if you want to keep it around - consider using `@Ignore("reason")`

Reply:
sorry, these were leftovers from me rewriting the typescript tests), deleted now.